torch/quantization/quantize.py

```diff
@@ -270,48 +271,59 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING, inplace=Fa
     convert(model, mapping, inplace=True)
     return model
 
-def quantize_dynamic(model, qconfig_dict=None, dtype=torch.qint8, mapping=DEFAULT_DYNAMIC_MODULE_MAPPING, inplace=False):
+def quantize_dynamic(model, qconfig=None, dtype=torch.qint8, mapping=DEFAULT_DYNAMIC_MODULE_MAPPING, inplace=False):
```

Comment:
Why not have an API with simpler meaning for qconfig and refactor the qconfig_dict creation logic into a seperate function? Currently qconfig can be a set of module names or just a dtype or a dictionary, which is confusing.


```
quantize_dynamic(model, torch.quantization.get_dynamic_qconfig_dict(quantization_set, dtype), mapping, inplace)
```
```python
# function to take in quantization set and dtype info and return a qconfig_dict
def get_dynamic_qconfig_dict(quantization_set=None, dtype=torch.qint8):
   if quantization_set is None:      
       if dtype == torch.qint8:
             qconfig = {
                nn.Linear : default_dynamic_qconfig,
                nn.LSTM : default_dynamic_qconfig,
            }
        elif dtype == torch.float16:
             qconfig = {
                # TODO: uncomment when float16 Linear support is added
                # nn.Linear : default_dynamic_qconfig,
                nn.LSTM : float16_dynamic_qconfig,
            }
        else:
            raise ValueError(
                "Don't know how to quantize with default settings for {}. Provide full qconfig please".format(dtype))
    elif isinstance(quantization_set, set):
        if dtype is torch.qint8:
            default_qconfig = default_dynamic_qconfig
        elif dtype is torch.float16:
            default_qconfig = float16_dynamic_qconfig
        else:
            raise RuntimeError('Unknown dtype specified for quantize_dynamic: ', str(dtype))
        qconfig = dict(zip(qconfig, itertools.repeat(default_qconfig)))
   return qconfig
```

Reply:
That looks way more complicated from an end user perspective...?

Reply:
The purpose of this PR is to *reduce* the amount of boilerplate the user has to write, what you've proposed does the exact opposite

Reply:
I think it is clearer from a end user perspective to just say:
```
quantize_dynamic(model, torch.quantization.get_dynamic_qconfig_dict(quantization_set, dtype)..)
```
rather than have qconfig_spec be a set or modules or dtypes or a dict. However, if you and @dzhulgakov  feel that this is ok, I will not object further.